models/agent_model.go

```diff
@@ -279,9 +279,19 @@ func (s *Agent) DSN(service *Service, dialTimeout time.Duration, database string
 			q.Set("connect_timeout", strconv.Itoa(int(dialTimeout.Seconds())))
 		}
 
+		address := ""
+		if socket == "" {
+			address = net.JoinHostPort(host, strconv.Itoa(int(port)))
+		} else {
+			// Set socket dirrectory as host URI parameter.
+			q.Set("host", socket)
+			// In case of empty url.URL.Host we need to identify a start of a path (database name).
+			database = "/" + database
```

Comment:
We need this as a host is empty: https://github.com/golang/go/blob/master/src/net/url/url.go#L833